Tests/ZcashPaymentURITests/MemoBytesTests.swift

```diff
@@ -58,4 +58,10 @@ final class MemoBytesTests: XCTestCase {
         
         XCTAssertThrowsError(try MemoBytes(bytes: [UInt8](repeating: 0xf4, count: 513)))
     }
+
+    func testInitWithInvalidTextFails() throws {
+        let invalidCharactersMemo = "QTw+Qg"
+
+        XCTAssertThrowsError(try MemoBytes(base64URL: invalidCharactersMemo))
+    }
```

Comment:
Add a test that uses all of the Base64URL characters (it should go with the other tests of success cases):
```swift
    /// Cross-check with Python's ``base64`` module using all Base64URL characters.
    /// ```python
    /// from base64 import b64decode, b64encode
    /// s = b"ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_"
    /// e = bytes([
    ///      0x00, 0x10, 0x83, 0x10, 0x51, 0x87, 0x20, 0x92, 0x8b, 0x30, 0xd3, 0x8f, 0x41, 0x14, 0x93, 0x51,
    ///      0x55, 0x97, 0x61, 0x96, 0x9b, 0x71, 0xd7, 0x9f, 0x82, 0x18, 0xa3, 0x92, 0x59, 0xa7, 0xa2, 0x9a,
    ///      0xab, 0xb2, 0xdb, 0xaf, 0xc3, 0x1c, 0xb3, 0xd3, 0x5d, 0xb7, 0xe3, 0x9e, 0xbb, 0xf3, 0xdf, 0xbf
    /// ])
    /// assert b64decode(s, altchars=b"-_") == e
    /// assert b64encode(e, altchars=b"-_") == s
    /// ```
    func testRoundTripWithAllBase64URLCharacters() {
        let s = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-_"
        let e: [UInt8] = [
            0x00, 0x10, 0x83, 0x10, 0x51, 0x87, 0x20, 0x92, 0x8b, 0x30, 0xd3, 0x8f, 0x41, 0x14, 0x93, 0x51,
            0x55, 0x97, 0x61, 0x96, 0x9b, 0x71, 0xd7, 0x9f, 0x82, 0x18, 0xa3, 0x92, 0x59, 0xa7, 0xa2, 0x9a,
            0xab, 0xb2, 0xdb, 0xaf, 0xc3, 0x1c, 0xb3, 0xd3, 0x5d, 0xb7, 0xe3, 0x9e, 0xbb, 0xf3, 0xdf, 0xbf
        ]

        let memo = try MemoBytes(base64URL: s)
        let memoFromBytes = try MemoBytes(bytes: e)

        XCTAssertEqual(memo, memoFromBytes)
        XCTAssertEqual(memo.toBase64URL(), s)
    }
```